h/admin/views/users.py

```diff
@@ -69,23 +63,56 @@ def users_activate(request):
                                     _query=(('username', user.username),)))
 
 
+@view_config(route_name='admin_users_rename',
+             request_method='POST',
+             permission='admin_users')
+def users_rename(request):
+    user = _form_request_user(request)
+
+    if user is None:
+        return httpexceptions.HTTPFound(
+            location=request.route_path('admin_users'))
+
+    old_username = user.username
+    new_username = request.params.get('new_username')
+
+    try:
+        svc = request.find_service(name='rename_user')
+        svc.check(old_username, new_username)
+
+        worker.rename_user.delay(old_username, new_username)
+
+        request.session.flash(
+            'The user "%s" will be renamed to "%s" in the backgroud. Refresh this page to see if it\'s already done' %
+            (old_username, new_username), 'success')
+
+        return httpexceptions.HTTPFound(
+            location=request.route_path('admin_users',
+                                        _query=(('username', new_username),)))
```

Comment:
This isn't about providing a user with the ability to rename their own account. This is an admin interface.

(And to answer your question, no. Any request that takes even 1 second is blocking other requests, so that's not an acceptable way to handle these kinds of changes.)